From 67746cee6561814c2bfe7566c6846ca4c6df906d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 2 Feb 2020 19:45:35 +0100
Subject: [PATCH 1/1] drivers/rng: simplify Kconfig

For all sandbox systems with DM_RNG we enable RNG_SANDBOX. So we can simply
set the default to yes.

All rng drivers depend on DM_RNG. Use a single 'if' instead of individual
dependencies. Now 'make menuconfig' shows the individual drivers neatly
indented under the DM_RNG entry.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 configs/sandbox64_defconfig | 1 -
 configs/sandbox_defconfig   | 1 -
 drivers/rng/Kconfig         | 9 +++++++--
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/configs/sandbox64_defconfig b/configs/sandbox64_defconfig
index 7b80033c3b..adc4a35fce 100644
--- a/configs/sandbox64_defconfig
+++ b/configs/sandbox64_defconfig
@@ -169,7 +169,6 @@ CONFIG_REMOTEPROC_SANDBOX=y
 CONFIG_DM_RESET=y
 CONFIG_SANDBOX_RESET=y
 CONFIG_DM_RNG=y
-CONFIG_RNG_SANDBOX=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_RV8803=y
 CONFIG_SANDBOX_SERIAL=y
diff --git a/configs/sandbox_defconfig b/configs/sandbox_defconfig
index fc018bdd2c..821975bd35 100644
--- a/configs/sandbox_defconfig
+++ b/configs/sandbox_defconfig
@@ -188,7 +188,6 @@ CONFIG_REMOTEPROC_SANDBOX=y
 CONFIG_DM_RESET=y
 CONFIG_SANDBOX_RESET=y
 CONFIG_DM_RNG=y
-CONFIG_RNG_SANDBOX=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_RV8803=y
 CONFIG_DEBUG_UART_SANDBOX=y
diff --git a/drivers/rng/Kconfig b/drivers/rng/Kconfig
index 893b89d49b..c1aa43b823 100644
--- a/drivers/rng/Kconfig
+++ b/drivers/rng/Kconfig
@@ -6,16 +6,21 @@ config DM_RNG
 	  This interface is used to initialise the rng device and to
 	  read the random seed from the device.
 
+if DM_RNG
+
 config RNG_SANDBOX
 	bool "Sandbox random number generator"
-	depends on SANDBOX && DM_RNG
+	depends on SANDBOX
+	default y
 	help
 	  Enable random number generator for sandbox. This is an
 	  emulation of a rng device.
 
 config RNG_STM32MP1
 	bool "Enable random number generator for STM32MP1"
-	depends on ARCH_STM32MP && DM_RNG
+	depends on ARCH_STM32MP
 	default n
 	help
 	  Enable STM32MP1 rng driver.
+
+endif
-- 
2.24.1

